packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1062,6 +1087,15 @@ export class Bucket extends BucketBase {
     this.lifecycleRules.push(rule);
   }
 
+  /**
+   * Adds server access logs to the bucket.
+   *
+   * @param serverAccessLogs configuration for the S3 bucket.
+   */
+  public addServerAccessLogs(serverAccessLogs: ServerAccessLogs): void {
+    this.serverAccessLogs = serverAccessLogs;
+  }
+
```

Comment:
Do we really need this mutating method? What's the use case of adding server access logs after the bucket is defined? In most cases we prefer all options to be defined during initialization because mutation is more susceptible to ordering issues. For example, what happens if two `addServerAccessLogs` are called with different options?

Reply:
I see, yea I think I'll better take it out